docs/js-guide/migrating.md

```diff
@@ -142,3 +143,68 @@ See [our approach to inheritance](https://github.com/dimagi/js-guide/blob/master
 [Sample pull request](https://github.com/dimagi/commcare-hq/pull/19938)
 
 Code that actually manipulates the prototype needs more thought.
+
+## Upgrading Select2
+
+HQ uses two different versions of select2. We're in the process of upgrading usage of the old version,
+because in principle it's crazy to have two versions but also because practically, having multiple versions
+makes it harder to bundle and share code across pages effectively.
+
+|                | New Select2                        | Old Select2                       |
+| -------------- | ---------------------------------- | --------------------------------- |
+| Version        |  4.0.0                             | 3.5.2                             |
+| Docs           | https://select2.org/               | http://select2.github.io/select2/ |
+| JS module      | `select2/dist/js/select2.full.min` | `select2-3.5.2-legacy/select2`    |
+| View decorator | `@use_select2_v4`                  | `@use_select2`                    |
+
+This is a fairly complicated migration both because the old and new version differ in multiple significant ways and because of how code is shared in HQ.
+
+### Figuring out what to migrate
+
+Migrating one select2 widget typically leads to migrating a number of other dependent ones. There are a couple of ways to pick a starting point:
+
+- Pick a view that uses the `use_select2` decorator
+- Pick a javascript module that directly depends on `select2-3.5.2-legacy/select2`
+- Pick a page that includes a script tag for `select2-3.5.2-legacy/select2.js` (rare, because most pages use the `@use_select2` decorator rather than including a script tag)
+- Browse the [requirejs bundle configuration](https://www.commcarehq.org/static/build.txt) to find modules that indirectly depend on `select2-3.5.2-legacy/select2`
+- Javascript modules that don't yet declare dependencies but that call the `.select2()` function
+- Pages that use a select2-dependent knockout binding like `select2` or `questionsSelect`
```

Comment:
yikes, that's quite a list

Reply:
yeah...this isn't a huge migration (I think there are about 50 instances) but finding all of the instances on any given page can be a pain